oss-internship-2020/pffft/main_pffft_sandboxed.cc

```diff
@@ -130,66 +139,65 @@ int main(int argc, char* argv[]) {
 
         if (max_iter_ == 0) max_iter_ = 1;
         if (cplx) {
-          api.cffti(N, wrk_.PtrBoth()).IgnoreError();
+          api.cffti(n, work_array.PtrBoth()).IgnoreError();
         } else {
-          api.rffti(N, wrk_.PtrBoth()).IgnoreError();
+          api.rffti(n, work_array.PtrBoth()).IgnoreError();
         }
         t0 = UclockSec();
 
-        for (iter = 0; iter < max_iter_; ++iter) {
+        for (int iter = 0; iter < max_iter_; ++iter) {
           if (cplx) {
-            api.cfftf(N, X_.PtrBoth(), wrk_.PtrBoth()).IgnoreError();
-            api.cfftb(N, X_.PtrBoth(), wrk_.PtrBoth()).IgnoreError();
+            api.cfftf(n, x_array.PtrBoth(), work_array.PtrBoth()).IgnoreError();
+            api.cfftb(n, x_array.PtrBoth(), work_array.PtrBoth()).IgnoreError();
           } else {
-            api.rfftf(N, X_.PtrBoth(), wrk_.PtrBoth()).IgnoreError();
-            api.rfftb(N, X_.PtrBoth(), wrk_.PtrBoth()).IgnoreError();
+            api.rfftf(n, x_array.PtrBoth(), work_array.PtrBoth()).IgnoreError();
+            api.rfftb(n, x_array.PtrBoth(), work_array.PtrBoth()).IgnoreError();
           }
         }
         t1 = UclockSec();
 
         flops =
-            (max_iter_ * 2) * ((cplx ? 5 : 2.5) * N * log((double)N) / M_LN2);
-        ShowOutput("FFTPack", N, cplx, flops, t0, t1, max_iter_);
+            (max_iter_ * 2) * ((cplx ? 5 : 2.5) * n * log((double)n) / M_LN2);
+        ShowOutput("FFTPack", n, cplx, flops, t0, t1, max_iter_);
       }
       
       // PFFFT benchmark
       {
         sapi::StatusOr<PFFFT_Setup*> s =
-            api.pffft_new_setup(N, cplx ? PFFFT_COMPLEX : PFFFT_REAL);
+            api.pffft_new_setup(n, cplx ? PFFFT_COMPLEX : PFFFT_REAL);
 
-        LOG(INFO) << "Setup status is: " << s.status().ToString().c_str()
-                  << "\n";
+        LOG(INFO) << "Setup status is: " << s.status().ToString();
 
         if (!s.ok()) {
           printf("Sandbox failed.\n");
-          return 1;
+          return EXIT_FAILURE;
         }
 
         sapi::v::RemotePtr s_reg(s.value());
```

Comment:
If you move the body of this function into `PffftMain()`, you can write
```c++
  SAPI_ASSIGN_OR_RETURN(PFFFT_Setup* setup,
                        api.pffft_new_setup(n, cplx ? PFFFT_COMPLEX : PFFFT_REAL));
  sapi::v::RemotePtr s_reg(setup);
```

Similar for the API calls below, where you `IgnoreError()`.